node/src/effect/requests.rs

```diff
@@ -190,86 +184,91 @@ where
 #[must_use]
-pub enum StorageRequest<S: StorageType + 'static> {
+pub enum StorageRequest {
     /// Store given block.
     PutBlock {
         /// Block to be stored.
-        block: Box<S::Block>,
+        block: Box<Block>,
         /// Responder to call with the result.  Returns true if the block was stored on this
         /// attempt or false if it was previously stored.
         responder: Responder<bool>,
     },
     /// Retrieve block with given hash.
     GetBlock {
         /// Hash of block to be retrieved.
-        block_hash: <S::Block as Value>::Id,
+        block_hash: BlockHash,
         /// Responder to call with the result.  Returns `None` is the block doesn't exist in local
         /// storage.
-        responder: Responder<Option<S::Block>>,
+        responder: Responder<Option<Block>>,
     },
     /// Retrieve block with given height.
     GetBlockAtHeight {
         /// Height of the block.
-        height: u64,
+        height: BlockHeight,
         /// Responder.
-        responder: Responder<Option<S::Block>>,
+        responder: Responder<Option<Block>>,
     },
     /// Retrieve highest block.
     GetHighestBlock {
         /// Responder.
-        responder: Responder<Option<S::Block>>,
+        responder: Responder<Option<Block>>,
     },
     /// Retrieve block header with given hash.
     GetBlockHeader {
         /// Hash of block to get header of.
-        block_hash: <S::Block as Value>::Id,
+        block_hash: BlockHash,
         /// Responder to call with the result.  Returns `None` is the block header doesn't exist in
         /// local storage.
-        responder: Responder<Option<<S::Block as Value>::Header>>,
+        responder: Responder<Option<BlockHeader>>,
     },
     /// Store given deploy.
     PutDeploy {
         /// Deploy to store.
-        deploy: Box<S::Deploy>,
+        deploy: Box<Deploy>,
         /// Responder to call with the result.  Returns true if the deploy was stored on this
         /// attempt or false if it was previously stored.
         responder: Responder<bool>,
     },
     /// Retrieve deploys with given hashes.
     GetDeploys {
         /// Hashes of deploys to be retrieved.
-        deploy_hashes: DeployHashes<S>,
+        deploy_hashes: Multiple<DeployHash>,
         /// Responder to call with the results.
-        responder: Responder<DeployResults<S>>,
+        responder: Responder<Vec<Option<Deploy>>>,
     },
     /// Retrieve deploy headers with given hashes.
     GetDeployHeaders {
         /// Hashes of deploy headers to be retrieved.
-        deploy_hashes: DeployHashes<S>,
+        deploy_hashes: Multiple<DeployHash>,
         /// Responder to call with the results.
-        responder: Responder<DeployHeaderResults<S>>,
+        responder: Responder<Vec<Option<DeployHeader>>>,
     },
-    /// Store the given execution results for the deploys in the given block.
+    /// Store a set of execution result for multiple deploys of a single block.
```

Comment:
"multiple" implies there must be more than one, which isn't the case.

Reply:
TIL, thanks!